packages/mui-material/src/Button/Button.js

```diff
@@ -477,6 +654,10 @@ Button.propTypes /* remove-proptypes */ = {
    * Element placed after the children.
    */
   endIcon: PropTypes.node,
+  /**
+   * Styles applied to the endIcon element if `loading={true}` and `loadingPosition="end"`.
+   */
+  endIconLoadingEnd: PropTypes.string,
```

Comment:
This prop shouldn't be here. These is part of the `classes` prop which is already handled in [buttonClasses.ts](https://github.com/mui/material-ui/pull/42987/files#diff-678f4576d82534ecac84d6c93e8166bad24d5d5aa804c61f4896853ca433631a)